### ISSUE TITLE
Incorrect info in Stochastic Gradient Descent lesson (MiniFlow)

### ISSUE THREAD

**Issue by georgesung** [udacity/sdc-issue-reports#133]
In the MiniFlow lesson module "14. Stochastic Gradient Descent", I noticed the following statement is incorrect:

The "stochastic" aspect of SGD refers to creating a random starting point for the weights and biases so that the loss begins at random point on the cost function's n-dimensional surface.

The "stochastic" part of SGD is that SGD stochastically/randomly chooses a subset of the entire training dataset to perform the gradient update procedure (and its dependencies, i.e. forward and backward passes). Technically, SGD chooses 1 datapoint/example from the training set to perform the gradient update, whereas (stochastic) "batch gradient descent" chooses a random batch of examples from the training set.

The lecture notes seem to be describing random parameter initialization, which is a separate topic from SGD.

Please let me know if my understanding is incorrect. Thanks.

**Comment by domluna** [comment]
@georgesung Nice catch! It's indeed "stochastic" because we're sampling data from the entire dataset randomly. I'll make an update shortly.

Also right that SGD is technically only choosing 1 example. But, it's much more common to sample a batch of examples than just 1 so SGD is typically used synonymously with batch stochastic gradient descent.

**Comment by domluna** [comment]
The proper definition is used now. Thanks again for the report.
